src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAroundCheck.java

```diff
@@ -901,6 +903,21 @@ private boolean isEmptyCtorBlock(DetailAST ast, int parentType) {
                     || isEmptyBlock(ast, parentType, TokenTypes.COMPACT_CTOR_DEF));
     }
 
+    /**
+     * Test if the given {@code DetailAST} is a part of a
+     * constructor checked from SLIST token.
+     *
+     * @param ast the {@code DetailAST} to test.
+     * @return {@code true} if {@code ast} makes up part of a
+     *          constructor block.
+     */
+    private static boolean isCtorBlockCheckedFromSlist(DetailAST ast) {
+        return ast.getType() == TokenTypes.SLIST
+                && (ast.getParent().getType() == TokenTypes.CTOR_DEF
+                || ast.getParent().getType() == TokenTypes.COMPACT_CTOR_DEF);
```

Comment:
@romani The pitest is failing but I think we would have to ignore it, reason-
Pitest is failing when `removed conditional - replaced equality check with true → SURVIVED` for both the lines 915 and 916
Now this will fail every time as when either one of them is replaced with `true` and it is a Constructor block, there are no other possibilities. What I mean is-
 ```
ast.getType() == TokenTypes.SLIST → true 
```
Then while visiting a `SLIST` token of a constructor, it can't be anything other than true, and this method is only concerned about constructors.

Reply:
Pitest failure can't be "ignored". It must either be suppressed with a reason why it can't be fixed or the code must be changed to be satisfied.

If I understand the argument, if the parent is always a CTOR, then the current AST is always a SLIST. If this is the case then it seems we can drop the SLIST check and keep the other 2. If you are unsure, you can always use regression to prove it won't create any differences. See https://github.com/checkstyle/checkstyle/issues/7797 step 3+ for how to do that if you are unfamiliar.